File: test_regress/t/t_c_this.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2021 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(simulator => 1);

compile();

if ($Self->{vlt_all}) {
    # The word 'this' (but only the whole word 'this' should have been replaced
    # in the contents.
    my $has_this   = 0;
    my $has_xthis  = 0;
    my $has_thisx  = 0;
    my $has_xthisx = 0;
    for my $file (glob_all("$Self->{obj_dir}/$Self->{VM_PREFIX}___024root__DepSet_*__0.cpp")) {
        my $text = file_contents($file);
        $has_this   = 1 if ($text =~ m/\bthis->clk\b/);
        $has_xthis  = 1 if ($text =~ m/\bxthis\b/);
        $has_thisx  = 1 if ($text =~ m/\bthisx\b/);
        $has_xthisx = 1 if ($text =~ m/\bxthisx\b/);
    }
    error("Some file has 'this->clk'") if $has_this;
    error("No file has 'xthis'") if !$has_xthis;
    error("No file has 'thisx'") if !$has_thisx;
    error("No file has 'xthisx'") if !$has_xthisx;
}

ok(1);
1;

File: test_regress/t/t_c_this.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('simulator')

test.compile()

if test.vlt_all:
    # The word 'this' (but only the whole word 'this' should have been replaced
    # in the contents.
    has_this = False
    has_xthis = False
    has_thisx = False
    has_xthisx = False
    for filename in test.glob_some(test.obj_dir + "/" + test.vm_prefix +
                                   "___024root__DepSet_*__0.cpp"):
        text = test.file_contents(filename)
        if re.search(r'\bthis->clk\b', text):
            has_this = True
        if re.search(r'\bxthis\b', text):
            has_xthis = True
        if re.search(r'\bthisx\b', text):
            has_thisx = True
        if re.search(r'\bxthisx\b', text):
            has_xthisx = True

    if has_this:
        test.error("Some file has 'this->clk'")
    if not has_xthis:
        test.error("No file has 'xthis'")
    if not has_thisx:
        test.error("No file has 'thisx'")
    if not has_xthisx:
        test.error("No file has 'xthisx'")

test.passes()
